src/borg/testsuite/archiver.py

```diff
@@ -2552,15 +2553,20 @@ def test_recreate_recompress(self):
         assert sha256_before == sha256_after
 
     def test_recreate_timestamp(self):
+        if sys.version_info >= (3, 6):
+            local_timezone = datetime.now(timezone(timedelta(0))).astimezone().tzinfo
+        else:
+            local_timezone = datetime.now(timezone.utc).astimezone().tzinfo
         self.create_test_files()
         self.cmd('init', '--encryption=repokey', self.repository_location)
         archive = self.repository_location + '::test0'
         self.cmd('create', archive, 'input')
         self.cmd('recreate', '--timestamp', "1970-01-02T00:00:00", '--comment',
                  'test', archive)
         info = self.cmd('info', archive).splitlines()
-        assert any([re.search(r'Time \(start\).+ 1970-01-02', item) for item in info])
-        assert any([re.search(r'Time \(end\).+ 1970-01-02', item) for item in info])
+        dtime = datetime(1970, 1, 2) + local_timezone.utcoffset(None)
+        assert any([re.search(r'Time \(start\).+ %s' % dtime.strftime("%Y-%m-%d"), item) for item in info])
+        assert any([re.search(r'Time \(end\).+ %s' % dtime.strftime("%Y-%m-%d"), item) for item in info])
```

Comment:
can you move the strftime call to outside the loops, only computing it once after line 2567?